### PR TITLE
Tools documentation

### LINKED ISSUE #359: Documentation and code description for Tools
## Expected Behavior

Code comments and documentation for introducing new Tools to backend. This will help contributors to easily create new Tools that could be later introduced in the graphical interface.

## Actual Behavior

Self-documenting code, not enough information to easily start creating new selectors.

### PR DESCRIPTION
**Pull request by kolszewska**
Fixes #359 

## Proposed Changes

  - Adde document explaining how to add new tool to MedTagger backend

### CODE REVIEW

File: backend/docs/introducing_new_tools.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+Adding new tool to MedTagger requires making changes both in backend and in frontend. To see how to add your tool on the frontend side proceed [here](). 
+
+### Step 1. Add tool to the definitions
+
+ Add your new tool to the `medtagger/definitions.py`. The `LabelTool` enum defines all tools supported by MedTagger.
+
+### Step 2. Create tool database model
+
+To enable support for various labels types we have introduced a concept of `LabelElement` which can be seen in `medtagger/database/models.py/`. It represents a high level LabelElement after which more specific types of LabelElement can inherit from. You can read more [here](http://docs.sqlalchemy.org/en/latest/orm/inheritance.html)
```
Comment (jszemplinski) [inline backend/docs/introducing_new_tools.md]: - various **label** types*
  - It represents a high level LabelElement, which more specific types of LabelElement can inherit from.

File: backend/docs/introducing_new_tools.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+#### Step 4.1 Repository
+
+Create a method in `medtagger/repositories/labels.py` that will create a new `db_session` and add your NewToolLabelElement. Please refer to other examples in the same file.
+
+#### Step 4.2 Business
+
+Create a method in `medtagger/api/scans/business.py` that will call method you have created in the last step. Please refer to other examples in the same file.
+
+Now search for `add_label_element` method in the same file. As you can see this method handless adding new label elements by the tool that the label was made with. Please add your tool and the method that will be responsible for creating such label.
```
Comment (jszemplinski) [inline backend/docs/introducing_new_tools.md]: handles*

File: backend/docs/introducing_new_tools.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+If your tool requires some additional validation. Please write the validation method and add it to `validate_label_payload` in `medtagger/api/scans/business.py`.
```
Comment (jszemplinski) [inline backend/docs/introducing_new_tools.md]: If (...) validation, please write (...)

File: backend/docs/introducing_new_tools.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+#### Step 4.3 REST Service
+
+As you can see in `medtagger/api/scans/service_rest.py` we do already have endpoint that adds new label for given scan. What is now important is to define a schema that will ensure that whatever JSON representation of a label made with your tool is correct.
```
Comment (jszemplinski) [inline backend/docs/introducing_new_tools.md]: (...) that will ensure that whatever JSON representation of a label made with your tool **is, it is correct**.

File: backend/docs/introducing_new_tools.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        "type": "object",
+        'oneOf': [
+            {'$ref': '#/definitions/rectangular_label_element_schema'},
+            {'$ref': '#/definitions/brush_label_element_schema'},
+            {'$ref': '#/definitions/point_label_element_schema'},
+            {'$ref': '#/definitions/chain_label_element_schema'},
+        ],
+    },
+```
+Below is and example of one of the schemas for label made with rectangle tool:
```
Comment (jszemplinski) [inline backend/docs/introducing_new_tools.md]: Below is **an** example (...)

File: backend/docs/introducing_new_tools.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+```
+
+Remember to make sure that your code passes linter tests. To do the testing, please run the following command from the `backend` directory: 
+```bash
+$ make tests    
+```
+
+If you want to know more about MedTagger's tests, go [here](testing.md)
+
+If all of your tests passes ,you are all done! You have succesfully added a new tool to MedTagger.
```
Comment (jszemplinski) [inline backend/docs/introducing_new_tools.md]: If all of your tests **pass,** you are all done!

File: backend/docs/introducing_new_tools.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+```
+Your job would be now to:
+
+1. Create your own definition of the schema,
+2. Write down every property that your `new_tool_label_element` should have,
+3. Provide additional information about properties (type, range etc.),
+4. Make sure to define which of the properties are required,
+5. Add your new definition to the `oneOf` section of the `elements_schema`.
+
+If you want more info about how to create such schemas, visit [this](https://json-schema.org/understanding-json-schema/index.html) page
```
Comment (jszemplinski) [inline backend/docs/introducing_new_tools.md]: Missing period ;)

File: backend/docs/introducing_new_tools.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
+Introducing new tools
+------------------------------------
+
+Here you can find every information that will be helpful while introducing new type of tools to backend. This tutorial assumes that you know how to run MedTagger. If not, go [here](development_setup_native.md) to learn how to run it natively, or [here](development_in_vagrant.md) to learn how to run it in Vagrant.
+
+Adding new tool to MedTagger requires making changes both in backend and in frontend. To see how to add your tool on the frontend side proceed [here]().
```
Comment (jszemplinski) [inline backend/docs/introducing_new_tools.md]: Empty link?

File: backend/docs/introducing_new_tools.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+
+### Step 1. Add tool to the definitions
+
+ Add your new tool to the `medtagger/definitions.py`. The `LabelTool` enum defines all tools supported by MedTagger.
+
+### Step 2. Create tool database model
+
+To enable support for various labels types we have introduced a concept of `LabelElement` which can be seen in `medtagger/database/models.py/`. It represents a high level LabelElement after which more specific types of LabelElement can inherit from. You can read more [here](http://docs.sqlalchemy.org/en/latest/orm/inheritance.html)
+
+As you can see from the `LabelElement` our `tool` column serves as the discriminator, which indicates the type of represented object:
```
Comment (jszemplinski) [inline backend/docs/introducing_new_tools.md]: As you can see **in the `LabelElement`, our `tool` column** serves (...)

File: backend/docs/introducing_new_tools.md (hunk 5 above)
Comment (jpowie01) [inline backend/docs/introducing_new_tools.md]: 1. `Below is and example of one...` -> Did you mean `Below is an example of one...`?
2. Please add a single empty line above this one.

File: backend/docs/introducing_new_tools.md [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+        'height': {'type': 'number', 'minimum': 0.0, 'maximum': 1.0},
+        'slice_index': {'type': 'integer'},
+        'tag': {'type': 'string'},
+        'tool': {'type': 'string', 'pattern': 'RECTANGLE'},
+    },
+    'required': ['x', 'y', 'width', 'height', 'slice_index', 'tag', 'tool'],
+    'additionalProperties': False,
+},
+```
+Your job would be now to:
```
Comment (jpowie01) [inline backend/docs/introducing_new_tools.md]: `Your job would be now to:` -> `Now you need to:`

File: backend/docs/introducing_new_tools.md [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+
+1. Create your own definition of the schema,
+2. Write down every property that your `new_tool_label_element` should have,
+3. Provide additional information about properties (type, range etc.),
+4. Make sure to define which of the properties are required,
```
Comment (jpowie01) [inline backend/docs/introducing_new_tools.md]: `which of the properties are required` -> `which of your properties are required`

File: backend/docs/introducing_new_tools.md [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+5. Add your new definition to the `oneOf` section of the `elements_schema`.
+
+If you want more info about how to create such schemas, visit [this](https://json-schema.org/understanding-json-schema/index.html) page
+
+### Step 5. Tests
+
+To make sure that all of this actually works, we need to write some tests! Proceed to `tests/functional_tests/test_adding_new_label.py`. This file holds tests for creating label for every tool that is supported by MedTagger. As you can see all of the tests have following steps:
```
Comment (jpowie01) [inline backend/docs/introducing_new_tools.md]: "Proceed to `tests/functional_tests/test_adding_new_label.py`. This file holds tests..." -> "Go to the `tests/functional_tests/test_adding_new_label.py` which contains all tests..."

File: backend/docs/introducing_new_tools.md [hunk 13]
```diff
@@ -0,0 +1,129 @@
...
+```bash
+$ make functional_tests
+```
+
+Remember to make sure that your code passes linter tests. To do the testing, please run the following command from the `backend` directory: 
+```bash
+$ make tests    
+```
+
+If you want to know more about MedTagger's tests, go [here](testing.md)
```
Comment (jpowie01) [inline backend/docs/introducing_new_tools.md]: Remember about period at the end of your sentence.

File: backend/docs/introducing_new_tools.md (hunk 6 above)
Comment (jpowie01) [inline backend/docs/introducing_new_tools.md]: 1. `passes ,you` -> `passes, you`.
2. `added a new tool` -> `added new tool`.

File: backend/docs/introducing_new_tools.md [hunk 14]
```diff
@@ -0,0 +1,129 @@
+Introducing new tools
+------------------------------------
```
Comment (jpowie01) [inline backend/docs/introducing_new_tools.md]: This underline should be the same length as above heading.

**Pull request merged**